src/scripts/migrate.py

```diff
@@ -96,25 +96,31 @@ def slurp(
     unmapped_terms: List[Term] = [x for x in owned_terms if x.curie not in sssom_object_ids]
     slurp_candidates = [x for x in unmapped_terms if x.curie not in excluded_terms]  # remove exclusions
 
-    # Determine slurpable terms
-    # Migratable term: A term owned by the given ontology but unmapped in Mondo, which has either no parents, or if it
-    # has parents, all of its parents have already been mapped in Mondo.
+    # Determine slurpable / migratable terms
+    # To be migratable, the term (i) must not already be mapped, (ii) must not be excluded (e.g. not in
+    # `reports/%_term_exclusions.txt`), and (iii) must not be deprecated / obsolete. Then, (iv) there are conditions
+    # concerning its parents. It must either (a) have no parents, or (b) have no valid parents in Mondo (i.e. all of its
+    # parent terms are marked obsolete in Mondo), or (c) all its parents must be mapped, and at least 1 of those
+    # parent's mappings must be one of `skos:exactMatch` or `skos:NarrowMatch`.
     terms_to_slurp: List[Dict[str, str]] = []
     for t in slurp_candidates:
-        # If all T.parents mapped, and 1+ is an exact or narrow match and non obsolete, designate T for slurping
-        # (i.e. only slurp if parents are already slurped)
+        obsolete_mondo_parent_ids = []
         qualified_parent_mondo_ids = []
+        all_parents_mapped = True
         no_parents: bool = not t.direct_owned_parent_curies
         for parent_curie in t.direct_owned_parent_curies:
-            # Conversely, if any of T.parents is unmapped, T is not to be slurped
+            if parent_curie not in sssom_object_ids and parent_curie in owned_term_curies:  # owned check redudnant?
+                all_parents_mapped = False
             if parent_curie in sssom_object_ids:
                 parent: Dict = sssom_df[sssom_df['object_id'] == parent_curie].to_dict('records')[0]
                 parent_mondo_id, parent_mondo_label = parent['subject_id'], parent['subject_label']
-                if not parent_mondo_label.startswith('obsolete'):
+                if parent_mondo_label.startswith('obsolete'):
+                    obsolete_mondo_parent_ids.append(parent_mondo_id)
+                elif parent['predicate_id'] in ['skos:exactMatch', 'skos:narrowMatch']:
                     qualified_parent_mondo_ids.append(parent_mondo_id)
-            if parent_curie not in sssom_object_ids and parent_curie in owned_term_curies:
-                break
-        if qualified_parent_mondo_ids or no_parents:
+
+        no_valid_mondo_parents: bool = len(t.direct_owned_parent_curies) == len(obsolete_mondo_parent_ids)
+        if (all_parents_mapped and qualified_parent_mondo_ids) or no_parents or no_valid_mondo_parents:
```

Comment:
I would suggest to refactor everything in the outer for loop into a method like:

```
_is_migration_candidate()
```

This isolates the logic more clearly from the rest of the pipeline, and makes it easier to maintain.

Also, add the documentation right into the code as comments as well. I will get lost if its just in the PR.

Reply:
`_is_migration_candidate()` makes sense, I will do that.

Regarding the documentation, I thought the same thing and did that already, if I understand what you mean. The same exact paragraph which describes what makes a term sluprable is in the `worklfows.md` as well as a code comment where the logic is. Although I will need to update it now that we have [simplified / removed the parent logic](https://github.com/monarch-initiative/mondo-ingest/pull/316#discussion_r1199661419).

Reply:
@matentzn You can hit `Resolve conversation` on this one and continue the discussion [here](https://github.com/monarch-initiative/mondo-ingest/pull/316#discussion_r1199681163) where I've added the aforementioned function.